app/Http/Controllers/HomeController.php

```diff
@@ -12,17 +12,52 @@ class HomeController extends Controller
     //
     public function index ()
     {
-        $tours = Tour::Limit(Config::get('limit.tours'))->inRandomOrder()->get();
-        $tour_new = Tour::Limit(Config::get('limit.new'))->orderBy('id', 'desc')->get();
+        $tours = Tour::limit(Config::get('limit.tours'))
+            ->inRandomOrder()->get();
+        $newTour = Tour::limit(Config::get('limit.new'))
+            ->orderBy('created_at', 'desc')->get();
 
-        return view('book_tour.pages.index', compact('tours', 'tour_new'));
+        return view('book_tour.pages.index', compact('tours', 'newTour'));
     }
 
     public function tours ()
     {
-    	$tours = Tour::Select('id', 'name', 'image', 'price', 'discount')
-    	    ->paginate(Config::get('paginate.tours'));
+        $tours = Tour::Select('id', 'name', 'image', 'price', 'discount')
+            ->orderBy('created_at', 'desc')
+            ->paginate(Config::get('paginate.tours'));
 
-    	return view('book_tour.pages.tours', compact('tours'));
+        return view('book_tour.pages.tours', compact('tours'));
+    }
+
+    public function redirectRoute ($route = null, $message = null)
+    {
+        return redirect()->route($route)->with($message);
+    }
+
+    public function tourDetail ($id)
+    {
+        $tour = Tour::find($id);
+        if ($tour) {
+            $type = Type::find($tour->type_id);
+            if ($type) {
+                $similarTours = $type->tours->except($tour->id)->toArray();
+
+                return view('book_tour.pages.tour_detail', compact('tour', 'similarTours'));
```

Comment:
Để retrun view trong cái if ($type) thế này lỡ cái $tour nó có mà cái ($type) không có thì mất luôn cái view?

Reply:
dạ trường hợp $tour có mà $type k có thì k thì không làm gì cả và trương trình tiếp tục thực thi : return ($this->redirectRoute('errors', ['message' => ''])); -> trả về một trang thông báo lỗi
vì trường hợp này chỉ có điều kiện if() thôi ạ

Reply:
Cái $type thì mục đích chỉ để lấy cái $similarTours.
Mà cái $similarTours có hay không đâu ảnh hưởng gì đến cái view $tour mà em redirect  nó ?
Cái $similarTours có thì hiển thị danh sách liên quan, k có thì khỏi hiện, ảnh hưởng gì đâu e